### ISSUE TITLE
Messages are lost because of [FORBIDDEN/8/index write (api)]

### ISSUE THREAD

**Issue by hc4** [Graylog2/graylog2-server#2211]
### Problem description

Just found, that some messages are lost.
It is clearly seen on chart:
![image](https://cloud.githubusercontent.com/assets/17475270/15146735/aa48b66a-16c5-11e6-828e-14ea1e7bf598.png)
During same time there is logged such error in log

```
2016-05-10T14:16:32.664+03:00 ERROR [Messages] Failed to index [500] messages. Please check the index error log in your web interface for the reason. Error: failure in bulk execution:
[0]: index [graylog2_81], type [message], id [a579a513-16a0-11e6-848a-005056b37533], message [ClusterBlockException[blocked by: [FORBIDDEN/8/index write (api)];]]
[1]: index [graylog2_81], type [message], id [a579a514-16a0-11e6-848a-005056b37533], message [ClusterBlockException[blocked by: [FORBIDDEN/8/index write (api)];]]
[2]: index [graylog2_81], type [message], id [a579a515-16a0-11e6-848a-005056b37533], message [ClusterBlockException[blocked by: [FORBIDDEN/8/index write (api)];]]
[3]: index [graylog2_81], type [message], id [a579a516-16a0-11e6-848a-005056b37533], message [ClusterBlockException[blocked by: [FORBIDDEN/8/index write (api)];]]
[4]: index [graylog2_81], type [message], id [a579a517-16a0-11e6-848a-005056b37533], message [ClusterBlockException[blocked by: [FORBIDDEN/8/index write (api)];]]
[5]: index [graylog2_81], type [message], id [a579cc20-16a0-11e6-848a-005056b37533], message [ClusterBlockException[blocked by: [FORBIDDEN/8/index write (api)];]]
[6]: index [graylog2_81], type [message], id [a579cc21-16a0-11e6-848a-005056b37533], message [ClusterBlockException[blocked by: [FORBIDDEN/8/index write (api)];]]
[7]: index [graylog2_81], type [message], id [a579cc22-16a0-11e6-848a-005056b37533], message [ClusterBlockException[blocked by: [FORBIDDEN/8/index write (api)];]]
[8]: index [graylog2_81], type [message], id [a579cc23-16a0-11e6-848a-005056b37533], message [ClusterBlockException[blocked by: [FORBIDDEN/8/index write (api)];]]
[9]: index [graylog2_81], type [message], id [a579cc24-16a0-11e6-848a-005056b37533], message [ClusterBlockException[blocked by: [FORBIDDEN/8/index write (api)];]]
[10]: index [graylog2_81], type [message], id [a579cc25-16a0-11e6-848a-005056b37533], message [ClusterBlockException[blocked by: [FORBIDDEN/8/index write (api)];]]
[11] ...
```

Another strange thing is that this errors not displayed on Indexer Failures page:
![image](https://cloud.githubusercontent.com/assets/17475270/15146789/f2f9a144-16c5-11e6-9021-8712c8e7c5fe.png)
### Environment
- Graylog Version:2.0 GA
- Elasticsearch Version:2.3.1

**Comment by hc4** [comment]
I've found, that in [Messages.java](https://github.com/Graylog2/graylog2-server/blob/4d4adfcdb074a6beeafcbee497c10e61dd84387f/graylog2-server/src/main/java/org/graylog2/indexer/messages/Messages.java) implemented retry logic in case of timeouts.
Is it possible to implement same logic in case of failures (like FORBIDDEN error)?
As I understand - FORBIDDEN error coused by ES node overflow and is not persistent.

**Comment by bernd** [comment]
This will be fixed with the upcoming Graylog 2.3 release where a ClusterBlockException triggers a retry.

**Comment by SamnangMao** [comment]
@hughbe

**Comment by bernd** [comment]
I am reopening this because we had to change retry behavior last minute before 2.3. This can still be an issue.

**Comment by SamnangMao** [comment]
@hughbe

**Comment by jhaar** [comment]
We just got hit by this - lost a tonne of data. We had our ES cluster run out of diskspace and the guys that fixed it didn't check beyond ES returning "GREEN". Only graylog seemed to think there was a problem, the graylog logfile was full of 

2018-04-09T20:47:45.112-07:00 WARN  [Messages] Failed to index message: index=<graylog_7446> id=<ed8f90a0-3c71-11e8-845c-a0369f951b70> error=<{"type":"cluster_block_exception","reason":"blocked by: [FORBIDDEN/8/index write (api)];"}>

To fix, all I had to do was force a "rotate write index" and immediately logging started showing up again - but it looks like we lost a tonne. Really weird - only seems to be data from our syslog-UDP Input channel  - our GELF and syslog-TLS Input data channels still showed up (which made this hard to notice)

This is graylog-2.4.3 and ES-5.6.8
